pygmt/clib/session.py

```diff
@@ -1068,6 +1069,77 @@ def put_matrix(self, dataset, matrix, pad=0):
         if status != 0:
             raise GMTCLibError(f"Failed to put matrix of type {matrix.dtype}.")
 
+    def read_data(
+        self,
+        family: str,
+        geometry: str,
+        mode: str,
+        wesn: Sequence[float] | None,
+        infile: str,
+        data=None,
+    ):
```

Comment:
The definition of the method follows the definition of the API function `GMT_Read_Data`. We need to call it like this:
```
lib.read_data("GMT_IS_DATASET", "GMT_IS_PLP", "GMT_READ_NORMAL", None, infile, None)
lib.read_data("GMT_IS_GRID", "GMT_IS_SURFACE", "GMT_READ_NORMAL", None, infile, None)
lib.read_data("GMT_IS_IMAGE", "GMT_IS_SURFACE", "GMT_READ_NORMAL", None, infile, None)
```
but they look really boring and weird.

I prefer to refactor the function definition like this:
```
lib.read_data(infile, kind="dataset")
lib.read_data(infile, kind="grid")
lib.read_data(infile, kind="image")
```
similar to the syntax of the `gmt read infile outfile -Tc|d|g|i|p|u` syntax of the special `read`/`write` module. For reference, the [`read` module](https://github.com/GenericMappingTools/gmt/blob/ff10eac4f78dd08b3adce4ac7f3b61430131ec82/src/gmtread.c#L192) mainly calls the [`gmt_copy` function](https://github.com/GenericMappingTools/gmt/blob/ff10eac4f78dd08b3adce4ac7f3b61430131ec82/src/gmt_api.c#L1294) and the `gmt_copy` function calls `GMT_Read_Data` with different arguments based on the data family.

Anyway, this should be discussed in more detail later.